pkg/generators/openapi.go

```diff
@@ -380,6 +413,9 @@ func (g openAPITypeWriter) generate(t *types.Type) error {
 				"},\n"+
 				"}\n}\n\n", args)
 			return nil
+		case hasV3OneOfTypes:
+			// having v3 oneOf types without custom v2 type or format does not make sense.
```

Comment:
Why not?

Reply:
Because if a type needs custom type definition, it at least should contain a "format" field. We require a type provide both "format" and "type" even if the type only wants one of them.

Reply:
And thus, the structure can either (1) do not use any type customization (2) provide all of v2 type definition and v3 oneOf